pinot-udf/src/test/java/org/apache/pinot/scalar/function/hypertrace/HypertraceScalarFunctionsTest.java

```diff
@@ -0,0 +1,39 @@
+package org.apache.pinot.scalar.function.hypertrace;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.hypertrace.core.attribute.service.projection.functions.Hash;
+import org.junit.jupiter.api.Test;
+
+public class HypertraceScalarFunctionsTest {
+  @Test
+  public void testHash() {
+    assertEquals(Hash.hash("foo"), HypertraceScalarFunctions.hash("foo"));
+    assertEquals("null", HypertraceScalarFunctions.hash("null"));
+    assertEquals("null", HypertraceScalarFunctions.hash(null));
+  }
+
+  @Test
+  public void testDefaultString() {
+    assertEquals("foo", HypertraceScalarFunctions.defaultString("foo", "bar"));
+    assertEquals("bar", HypertraceScalarFunctions.defaultString("", "bar"));
+    assertEquals("bar", HypertraceScalarFunctions.defaultString("null", "bar"));
+    assertEquals("null", HypertraceScalarFunctions.defaultString("", "null"));
+  }
+
+  @Test
+  public void testStringEquals() {
+    assertEquals("true", HypertraceScalarFunctions.stringEquals("foo", "foo"));
+    assertEquals("false", HypertraceScalarFunctions.stringEquals("foo", ""));
+  }
+
+  @Test
+  public void testConditional() {
+    assertEquals("foo", HypertraceScalarFunctions.conditional("true", "foo", "bar"));
+    assertEquals("foo", HypertraceScalarFunctions.conditional("TRuE", "foo", "bar"));
+    assertEquals("bar", HypertraceScalarFunctions.conditional("FALSE", "foo", "bar"));
+    assertEquals("null", HypertraceScalarFunctions.conditional("null", "foo", "bar"));
+    assertEquals("bar", HypertraceScalarFunctions.conditional(null, "foo", "bar"));
```

Comment:
I'd expect this one to be null - it shouldn't behave differently than the one above on line 35.

Reply:
So you also want to check for value `null` as well instead of passing it on the `parseBoolean`?

Reply:
Pinot doesn't ever pass us null, so it's not an important thing to test. I'd recommend removing it entirely, but if do we keep the test we should make sure it's behavior matches pinot's null string, since they semantically mean the same.

Reply:
done. kept it and changed the test